lib/elixir_tic_tac_toe_basic.ex

```diff
@@ -3,24 +3,37 @@ defmodule ElixirTicTacToeBasic do
   This module starts the game.
   """
 
-  defstruct [:ui, :player]
+  defstruct [:ui, :player, :outcome_checker]
 
-  def start(%{ui: ui, player: player} = config) do
+  def start(config) do
     config
     |> start_game()
-    |> take_turn()
-    |> take_turn()
+    |> play_game()
+    |> end_game()
   end
 
-  defp start_game(%{ui: ui, player: player} = config) do
+  defp start_game(%{ui: ui} = config) do
     config
     |> ui.welcome()
     |> ui.display_board()
   end
 
+  defp play_game(%{outcome_checker: outcome_checker} = config) do
+    config = take_turn(config)
+
+    is_full = outcome_checker.check_full(config)
```

Comment:
**suggestion (non-blocker)**: Consider changing this function name to reveal that the game is over.

`check_full` reveals an implementation detail, which is that the game is over when the board is full. However, the idea that you are trying to communicate is that the game is over.